### PR TITLE
Tests and housekeeping

### PR DESCRIPTION
**Pull request by nblumhardt**
**To do:** Need to test the resulting app package :-)

 * Adds a unit test project - some solution reorganization and build script updates were required for this
 * Split helper methods out of `SlackReactor` for easier testing
 * Updated the icon URL to an equivalent but different one so that Slack refreshes existing installs
 * Added some classes to model the Slack API's contract types (replacing the anon. types that were used)
 * Brief **Getting started** instructions in the README
 * Trivial ReSharper-directed tidy-up (`var` in a few places, etc.)
 * Default the app name to the instance name within Seq, now that this is available

### CODE REVIEW

**Comment by nblumhardt** [comment]
CC @KodrAus - some extra eyes on this would be great :-)

**Comment by nblumhardt** [comment]
image

Works like a charm.

File: README.md [hunk 1]
```diff
@@ -2,3 +2,12 @@
 [![NuGet](https://img.shields.io/nuget/v/Seq.App.Slack.svg?style=flat-square)](https://www.nuget.org/packages/Seq.App.Slack/)
 
 An app for Seq (http://getseq.net) that forwards messages to Slack.
+
+### Getting started
+
+ 1. Install the app into Seq through the Seq UI: _Settings_ > _Apps_ > _Install from Nuget_. The package id is _Seq.App.Slack_.
+ 2. In Slack, select _Manage apps_ > _Search App Directory_ > _"Incoming WebHooks"
```
Comment (KodrAus) [inline README.md]: Missing `_` at the end here?

**Comment by nblumhardt** [comment]
I need to do a little more work to get tests running in CI, but I'm out of time for now. I'll shepherd what we have, through, and pick this up again as soon as I get the chance :-)

**Pull request merged**